runway/blueprints/staticsite/auth_at_edge.py

```diff
@@ -0,0 +1,434 @@
+                                description,   # type: str
+                                handler,  # type: str
+                                role  # type: iam.Role
+                                ):
+        # type: (...) -> awslambda.Function
+        """Create an Auth@Edge lambda resource.
+
+        Args:
+            name (str): The name of the function in PascalCase
+            description (str): Description to be displayed in the
+                lambda panel
+            handle (str): The underscore separated representation
+                of the name of the lambda. This handle is used to
+                determine the handler for the lambda as well as
+                identify the correct Code hook_data information.
+            role (IAM.Role): The Lambda Execution Role
+        """
+        return self.template.add_resource(
+            awslambda.Function(
+                title,
+                Code=self.context.hook_data['aae_lambda_config'].get(handler),
+                Description=description,
+                Handler='__init__.handler',
+                Role=role.get_att("Arn"),
+                Runtime='python3.7'
+            )
+        )
+
+    def add_version(self,
+                    title,  # type: str
+                    lambda_function  # type: awslambda.Function
+                    ):
+        # type: (...) -> awslambda.Version
+        """Create a version association with a Lambda@EDGE function.
+
+        In order to ensure different versions of the function
+        are appropriately uploaded a hash based on the code of the
+        lambda is appended to the name. As the code changes so
+        will this hash value.
+
+        Args:
+            title (str): The name of the function in PascalCase
+            role (awslambda.Function): The Lambda function
+        """
+        code_hash = hashlib.md5(
+            str(lambda_function.properties['Code'].to_dict()).encode()
+        ).hexdigest()
+
+        return self.template.add_resource(
+            awslambda.Version(
+                title + 'Ver' + code_hash,
+                FunctionName=lambda_function.ref()
+            )
+        )
+
+    def add_lambda_execution_role(self):
+        # type: () -> iam.Role
+        """Create the Lambda@Edge execution role."""
+        return self.template.add_resource(
+            iam.Role(
+                'LambdaExecutionRole',
+                AssumeRolePolicyDocument=PolicyDocument(
+                    Version='2012-10-17',
+                    Statement=[
+                        Statement(
+                            Effect=Allow,
+                            Action=[awacs.sts.AssumeRole],
+                            Principal=Principal('Service',
+                                                ['lambda.amazonaws.com',
+                                                 'edgelambda.amazonaws.com'])
+                        )
+                    ]
+                ),
```

Comment:
awacs has a helper to simplify this. will need to `from awacs.helpers.trust import make_simple_assume_policy`

```suggestion
                AssumeRolePolicyDocument=make_simple_assume_policy(
                    'lambda.amazonaws.com', 'edgelambda.amazonaws.com'
                ),
```

Reply:
Updated